initfiles/componentfiles/thor/stop_thor

```diff
@@ -47,23 +47,17 @@ if [ "$#" -lt "2" ] || [ "$2" != "keep_sentinel" ]; then
     sleep 1
 fi
 
-masterpid=`cat ${MASTER_PID_NAME} 2> /dev/null`
-if [ ! -z $masterpid ]; then
-  while :
-  do
-    kill -0 $masterpid >& /dev/null
-    masterRunning=$(( $? == 0 ? 1 : 0 ))
-    if [ 0 == $masterRunning ]; then
-      break
-    fi
-    echo --------------------------
-    echo stopping thormaster $THORMASTER 
-    kill $masterpid >& /dev/null
-    sleep 8
-    kill -9 $masterpid >& /dev/null
-    sleep 1
-  done
+masterpid=$( cat $MASTER_PID_NAME )
+TIMEOUT=8000
+kill $masterpid &>/dev/null
+while [ -d /proc/$masterpid -a $TIMEOUT -gt 0 ]; do
+  sleep 0.2
+  TIMEOUT=$(($TIMEOUT-200))
+done
+if [ $TIMEOUT -le 0 ]; then
+  kill -9 $masterpid &>/dev/null
```

Comment:
The previous code would still loop around after a kill -9 ...
which seems a bit pointless in general.
However, it can happen I think, if the process gets into a zombie state..

So should probably detect if kill -9 fails to kill, report it and abort here.